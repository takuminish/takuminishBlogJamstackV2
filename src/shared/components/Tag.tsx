import * as React from "react";
import Box from "@mui/material/Box";

type Props = {
  text: string;
};

function Tag(props: Props) {
  const { text } = props;

  return (
    <Box
      component="span"
      sx={{
        padding: "0 7px",
        backgroundColor: "#94a3b8",
        borderRadius: "10px",
        color: "#ffffff",
        fontSize: "13px",
        fontWeight: 700,
      }}
    >
      {text}
    </Box>
  );
}

export default Tag;
