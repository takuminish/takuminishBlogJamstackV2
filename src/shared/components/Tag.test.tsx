import React from "react";
import Tag from "./Tag";
import { render, screen } from "@testing-library/react";

const createDOM = () => render(<Tag text="test" />);

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});

test("props textで与えられた文字列が表示されていること", () => {
  createDOM();

  screen.findAllByText("test");
});
