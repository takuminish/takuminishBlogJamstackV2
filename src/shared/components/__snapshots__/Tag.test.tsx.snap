// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <span
    class="MuiBox-root css-1kcf2o3"
  >
    test
  </span>
</div>
`;
